Content.Server/Connection/ConnectionManager.IPIntel.cs

```diff
@@ -0,0 +1,226 @@
+
+        if (request.StatusCode == HttpStatusCode.TooManyRequests)
+        {
+            // TODO: Ok so nik did kinda kick start my brain that... well... this could either be a daily limit or a minute limit and i would have no idea...
+            // I was gonna treat this as daily but yeah i see the issue here...
+            // We are just gonna try to estimate our ratelimit
+            //... maybe wait a minute send the request again and if we are still ratelimited assume we hit the daily limit and cry
+            // also helps to have minute/day requests stored in the db
+
+            _sawmill.Warning("We hit the IPIntel request limit at some point.");
+            return _rejectLimited ? (true, Loc.GetString("ipintel-server-ratelimited")) : (false, string.Empty);
+        }
+
+        var response = await request.Content.ReadAsStringAsync();
+
+        switch (response)
+        {
+            case "-1":
+            {
+                _sawmill.Error("IPIntel returned error -1: Invalid/No input.");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            case "-2":
+            {
+                _sawmill.Error("IPIntel returned error -2: Invalid IP address.");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            case "-3":
+            {
+                _sawmill.Error("IPIntel returned error -3: Unroutable address / private address given to the api.");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            case "-4":
+            {
+                _sawmill.Error("IPIntel returned error -4: Unable to reach IPIntel database. Perhaps it's down?");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            case "-5":
+            {
+                _sawmill.Error("IPIntel returned error -5: Server's IP/Contact may have been banned, go to getipintel.net and make contact to be unbanned.");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            case "-6":
+            {
+                _sawmill.Error("IPIntel returned error -6: You did not provide any contact information with your query or the contact information is invalid.");
+                return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+            }
+            default:
+            {
+                return await ScoreCheck(response, e, expired, query.Count == 0);
+            }
+        }
+    }
+
+    private async Task HandleRatelimit()
+    {
+        // Oh my god this is terrible
+        if (_currentRequestsDay < _requestLimitDay)
```

Comment:
Doesn't this code not check the minute rate limit until the day rate limit fills up? Because it's in the else block.